src/clj/puppetlabs/puppetserver/certificate_authority.clj

```diff
@@ -722,7 +722,10 @@
                          #{:key-encipherment
                            :digital-signature} true)
                        (utils/subject-key-identifier
-                         master-public-key false)]]
+                         master-public-key false)
+                       {:oid cli-auth-oid
+                        :critical false
+                        :value "true"}]]
```

Comment:
Would we prefer this as a utility method in `jvm-ssl-utils` instead? It seemed like maybe overkill given how simple it is, but I'm happy to make one.

Reply:
I don't think that putting it in ssl-utils is necessary though we may want to put an assertion in `initialize-master-ssl!-test` to call out how important it is and make sure it round trips like we expect?

Reply:
Will do. Didn't want to dig into testing until I knew that this was an okay approach.